### PR TITLE
Implement a Firehose -> Stackdriver nozzle

### PR DESCRIPTION
**Pull request by cholick**
This MVP connects to the firehose and sends HTTPStartStop events to Stackdriver.

Non-goals to be implemented (or not) later:
- Error handling
- Filtering of event types
- More configuration
- Mapping from events to labels?
- Alternate GCP authentication

### CODE REVIEW

File: src/stackdriver-nozzle/firehose/client.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+func (fc *client) StartListening(nozzle logging.Logging) error {
+	cfClient := cfclient.NewClient(fc.cfConfig)
+
+	cachingClient := caching.NewCachingEmpty()
+
+	eventRouter := eventRouting.NewEventRouting(cachingClient, nozzle)
+
+	//err := eventRouter.SetupEventRouting("LogMessage,ValueMetric,HttpStartStop,CounterEvent,Error,ContainerMetric")
```
Comment (johnsonj) [inline src/stackdriver-nozzle/firehose/client.go]: is there anyway to specify * here? I'm assuming this commented line is a helper for later when you add all event types

File: src/stackdriver-nozzle/main.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+	if *debug {
+		println("Sending firehose to standard out")
+		err := client.StartListening(&dev.StdOut{})
+		if err != nil {
+			panic(err)
+		}
+	} else {
+		println("Sending firehose to Stackdriver")
+		sdClient := stackdriver.NewClient(*projectID)
+		//sdClient.Post("hello world 5")
```
Comment (johnsonj) [inline src/stackdriver-nozzle/main.go]: remove

File: src/stackdriver-nozzle/main.go [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+				Default("false").
+				OverrideDefaultFromEnvar("SKIP_SSL_VALIDATION").
+				Bool()
+	projectID = kingpin.Flag("project-id", "gcp project id").
+			OverrideDefaultFromEnvar("PROJECT_ID").
+			String() //maybe we can get this from gcp env...? research
+)
+
+func main() {
+	//todo: pull in logging library...
```
Comment (johnsonj) [inline src/stackdriver-nozzle/main.go]: is this for the logging of the nozzle?

File: src/stackdriver-nozzle/nozzle/nozzle.go [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+type Nozzle struct {
+	StackdriverClient stackdriver.Client
+}
+
+func (n *Nozzle) Connect() bool {
+	return true
+}
+
+func (n *Nozzle) ShipEvents(event map[string]interface{}, _ string /* TODO research second string */) {
```
Comment (johnsonj) [inline src/stackdriver-nozzle/nozzle/nozzle.go]: curious what you have planned for the second string? maybe remove it for now

File: src/stackdriver-nozzle/main.go (hunk 3 above)
Comment (nrxus) [inline src/stackdriver-nozzle/main.go]: We have straight up print statements in here right now, but we really should use a real logging library so we can filter them by debug/info/etc...

File: src/stackdriver-nozzle/nozzle/nozzle.go (hunk 4 above)
Comment (nrxus) [inline src/stackdriver-nozzle/nozzle/nozzle.go]: The interface specifies the second string - it turns out to be a log message, but we haven't implemented any events that actually use it yet.

File: src/stackdriver-nozzle/firehose/client.go (hunk 1 above)
Comment (nrxus) [inline src/stackdriver-nozzle/firehose/client.go]: It doesn't look like it from the source, but we can create the full list ourselves from here:

https://github.com/cloudfoundry/sonde-go/blob/master/events/envelope.pb.go#L61-L76

When we implement the config for filtering, it should become moot, I think?

**Pull request merged**